astrokat/scans.py

```diff
@@ -146,15 +214,108 @@ def reversescan(session, target, nd_period=None, lead_time=None, **kwargs):
+            key = 'radec_p%i' % i
+            if key in kwargs:
+                target_list.append(
+                    katpoint.Target(
+                        't%i,radec,%s' % (i, kwargs[key]),
+                        antenna=target.antenna,
+                    )
+                )
+    else:
+        user_logger.error("No scan area defined - require radec_p1 and radec_p2")
+        return False
+    direction = kwargs.get("direction", False)
+    scan_speed = kwargs.get("scan_speed", _DEFAULT_SCAN_SPEED_ARCMIN_PER_SEC)
+
+    obs_start_ts = katpoint.Timestamp(time.time()).to_ephem_date()
+    # use 1 deg offset to pre-position >4 min in the future to take into account slewing
+    el, az_min, az_max, t_start, t_end = _get_scan_area_extents(target_list, antenna,
+                                                                obs_start_ts,
+                                                                offset_deg=1)
+
+    # TODO: get horizon limit from observation - may want to pass limits of
+    #       the "acceptable elevation extent" into _get_scan_area_extents.
+    if _MIN_HORIZON_EL_FOR_SCAN_DEG > np.degrees(el):
+        user_logger.warning(
+            "Source and scan below horizon: %s < %s",
+            np.degrees(el),
+            _MIN_HORIZON_EL_FOR_SCAN_DEG,
+        )
+        return False
+
+    scan_target = katpoint.construct_azel_target(katpoint.wrap_angle(az_min), el)
+    scan_target.name = target.name
+    # katpoint destructively set dates and times during calculation
+    # restore datetime before continuing
+    scan_target.antenna = antenna
+    scan_target.antenna.observer.date = obs_start_ts
+    user_logger.info("Slew to scan start")  # slew to target.
+    target_visible = session.track(scan_target, duration=0.0, announce=False)
+    if not target_visible:
+        user_logger.warning(
+            "Start of scan is not visible!  Elevation: %.1f deg.", np.degrees(el)
+        )
+        return False
+
+    # This is the real scan
+    obs_start_ts = katpoint.Timestamp(time.time()).to_ephem_date()
+    el, az_min, az_max, t_start, t_end = _get_scan_area_extents(target_list, antenna,
+                                                                obs_start_ts)
+    scan_target = katpoint.construct_azel_target(
+        katpoint.wrap_angle((az_min + az_max) / 2.), el)
+    scan_target.name = target.name
+    # katpoint destructively set dates and times during calculation
+    # restore datetime before continuing
+    scan_target.antenna = antenna
+    scan_target.antenna.observer.date = obs_start_ts
+
+    scan_start = np.degrees(az_min - (az_min + az_max) / 2.)
+    scan_end = np.degrees(az_max - (az_min + az_max) / 2.)
+
+    scanargs = {}
+    if "projection" in kwargs:
+        scanargs["projection"] = kwargs["projection"]
+
+    # take into account projection effects of the sky and convert to degrees per second
+    # 5 arcmin/s should translate to 5/cos(el)
```

Comment:
These comments on these two lines are a little confusing for me. 
The `5/cos(el)` takes the projection into account, but not the conversion from arcmin to degrees.  

Maybe:

```python
    # take into account projection effects of the sky and convert to degrees per second
    # E.g., 5 arcmin/s should translate to 5/60/cos(el) deg/s
```

Reply:
Done